sdk/tests/conformance/reading/read-pixels-test.html

```diff
@@ -124,80 +126,128 @@
     }
   }
 
-  var badFormats = [
+  continueTestPart2();
+}
+
+function continueTestPart2() {
+  var invalidFormat = [gl.DEPTH_COMPONENT, gl.DEPTH_STENCIL, desktopGL.R8, gl.RGBA4];
+  if (contextVersion < 2) {
+  // They are valid in WebGL 2 or higher
+    invalidFormat = invalidFormat.concat([gl.LUMINANCE, gl.LUMINANCE_ALPHA, desktopGL.RED, desktopGL.RG_INTEGER, desktopGL.RGBA_INTEGER]);
+  }
+
+  var invalidTypeInfo = [
+    {type: gl.UNSIGNED_SHORT,                 dest: new Uint16Array(4)},
+    {type: gl.SHORT,                          dest: new Int16Array(4)},
+    {type: desktopGL.UNSIGNED_INT_24_8,       dest: new Uint32Array(4)}
+  ];
+  if (contextVersion < 2) {
+  // They are valid in WebGL 2 or Higher
+    invalidTypeInfo = invalidTypeInfo.concat([
+      {type: gl.BYTE,                               dest: new Int8Array(4)},
+      {type: gl.UNSIGNED_INT,                       dest: new Uint32Array(4)},
+      {type: desktopGL.UNSIGNED_INT_2_10_10_10_REV, dest: new Uint32Array(4)},
+      {type: gl.INT,                                dest: new Int32Array(4)}
+    ]);
+  }
+
+  debug("");
+  debug("check invalid format or type");
+  for (var ff = 0; ff < invalidFormat.length; ++ff) {
+    var format = invalidFormat[ff];
+    var buf = new Uint8Array(4);
+    gl.readPixels(0, 0, 1, 1, format, gl.UNSIGNED_BYTE, buf);
+    wtu.glErrorShouldBe(gl, gl.INVALID_ENUM, "Should not be able to read as " + wtu.glEnumToString(gl, format));
+  }
+
+  for (var tt = 0; tt < invalidTypeInfo.length; ++tt) {
+    var info = invalidTypeInfo[tt];
+    var type = info.type;
+    var dest = info.dest;
+    gl.readPixels(0, 0, 1, 1, gl.RGBA, type, dest);
+    wtu.glErrorShouldBe(gl, gl.INVALID_ENUM, "Should not be able to read as " + wtu.glEnumToString(gl, type));
+  }
+
+  var badCombinations = [
     {
       format: gl.RGB,
```

Comment:
You haven't checked the GL_IMPLEMENTATION_COLOR_READ_FORMAT and GL_IMPLEMENTATION_COLOR_READ_TYPE yet, how can you be sure this is invalid combination? The same question for a few other combinations below.